c10/cuda/CUDAStream.h

```diff
@@ -155,10 +155,11 @@ class C10_CUDA_API CUDAStream {
 
   static std::tuple<int, int> priority_range() {
     #ifndef __HIP_PLATFORM_HCC__
-      int least_priority, greatest_priority;
-      C10_CUDA_CHECK(
-        cudaDeviceGetStreamPriorityRange(&least_priority, &greatest_priority));
-      return std::make_tuple(least_priority, greatest_priority);
+      // Note: this returns the range of priority **supported by PyTorch**, not
+      // the range of priority **supported by CUDA**. The former is a subset of
+      // the latter. Curently PyTorch only supports 0 and -1, which are "low" and
+      // "high" priority.
+      return std::make_tuple(0, -1);
```

Comment:
cc: @ngimel @ptrblck

Reply:
Without this change, the test will fail with
```
Jun 26 22:34:46 ======================================================================
Jun 26 22:34:46 FAIL [0.030s]: test_streams_priority (__main__.TestCuda)
Jun 26 22:34:46 ----------------------------------------------------------------------
Jun 26 22:34:46 Traceback (most recent call last):
Jun 26 22:34:46   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 777, in wrapper
Jun 26 22:34:46     method(*args, **kwargs)
Jun 26 22:34:46   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 777, in wrapper
Jun 26 22:34:46     method(*args, **kwargs)
Jun 26 22:34:46   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 387, in wrapper
Jun 26 22:34:46     fn(*args, **kwargs)
Jun 26 22:34:46   File "test_cuda.py", line 946, in test_streams_priority
Jun 26 22:34:46     self.assertEqual(high, s1.priority)
Jun 26 22:34:46   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 1122, in assertEqual
Jun 26 22:34:46     self.assertTrue(result, msg=msg)
Jun 26 22:34:46 AssertionError: False is not true : Scalars failed to compare as equal! Comparing -2 and -1 gives a difference of 1, but the allowed difference with rtol=0 and atol=0 is only 0!
Jun 26 22:34:46 
Jun 26 22:34:46 ----------------------------------------------------------------------
```

Reply:
@zasdfgbnm hmm, perhaps we have to modify the test rather than CUDAStream.h

Reply:
Perhaps we should leave the API call here and assert that least priority is greater or equal to 0, and greatest priority is less or equal to -1, then creating the tuple with hardcoded values is safe. Having just 2 possible priorities is hardcoded deep in the pytorch.

Reply:
fixed